File: 3Star BegLvl/BEUTGAR.py
def garlandBeautiful():
    s = input()
    n = len(s)
    cr = cp = cg = 0
    flag = False
    if n % 2 != 0:
        flag = True
    for i in range(n):
        if s[i] == 'R':
            cr += 1
        else:
            cg += 1
        if i > 0:
            if s[i - 1] == s[i]:
                cp += 1
    if n > 2 and s[0] == s[n - 1]:
        cp += 1
    if cr != cg or cp > 2:
        flag = True
    if flag:
        print("no")
    else:
        print("yes")

t = int(input())
for _ in range(t):
    garlandBeautiful()

File: 3Star BegLvl/BEUTGAR.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long

void garlandBeautiful() {
    string s;
    cin >> s;
    int n = s.size();
    ll cr=0, cp=0, cg=0;
    bool flag = false;
    if (s.size()%2)
        flag=true;
    for (int i=0; i<n; i++) {
        if (s[i]=='R')
            cr++;
        else
            cg++;
        if (i>0) {
            if (s[i-1]==s[i])
                cp++;
        }
	}
    if (s.size()>2 && s[0]==s[s.size()-1])
        cp++;
    if (cr!=cg || cp>2)
        flag = true;
    if (flag)
        cout << "no" << endl;
    else
        cout << "yes" << endl;
}

int main() {
	// your code goes here
	int t;
	cin >> t;
	while (t--) {
	    garlandBeautiful();
	}
	return 0;
}
